x-pack/plugins/security_solution/common/endpoint/types/actions.ts

```diff
@@ -271,6 +271,11 @@ export interface ActionDetails<TOutputContent extends object = object> {
    * This is an Array because the action could have been sent to multiple endpoints.
    */
   agents: string[];
+  /**
+   * The endpoint ID/fleet agent ID which was used to create the action (same as above) and the agent's name
+   * Action could have been sent for multiple endpoints which is why this is an Array
+   */
+  hosts?: Record<string, { name: string }>;
```

Comment:
I mentioned earlier that the Details API needs to return this data. For the details, just liike `outputs`, this value should always be included (not optional), so this property here needs to have its `?` removed.

See my comment below for what shoudl be done for the List API response type

Reply:
Okay, I'm a bit confused now! 

I thought you suggested that we use the optional parameter so that we return hosts (names) for page-level API requests where it is needed. As I understood earlier, we want to show hostnames only on the response actions page and so the actions list API should return it only when it's requested via that page. 

Responder and endpoint details action logs don't show the hostnames and hence shouldn't get this from the API response.

Maybe I missed something during the back and forth. 😅

Reply:
We have two api: a details API and a List API. The details API is not as expensive as the List api because it does not loop through all records. So on the details API, we should return "all" data for the record. Example: if someone today wants to look at the `outputs` of an action, they will need to call the details, because we exclude it from the List API.

The same should apply for `hosts`. It should (IMO) be included in the Details, but conditionally include on the list API response only if the URL param is set.

If don't do that change (which I guess I'm ok with as well, if we all agree), then the types need to be changed so that its clear what comes for each API. If we go this route, then `ActionDetails` should never show an optional `hosts`, and that should only appear on the return value from the List api.  We could also diverge the types and create two different ones, but I think that is unnecessary and would likely cause even more confusing with the Actions stuff we already have on the api side)

_____

One other thing I forgot to mention on my review: I noticed that we are making only one API call into the Metadata API with all of the action IDs (in the current page of data), which is awesome. Should be much more performant. My initial thought was that we were going to make multiples - one for each Action, so this is good.  And now I'm maybe back to: should we just always include it in the response for both Detail and List apis? 🙈

Reply:
😆  So I just assumed that making a request to metadata among all the endpoint indices was expensive, which is why we're doing this optional thing. If it is not that expensive then it does make sense to make the `hosts` be present in both API responses. That would go towards keeping the same shape of the response everywhere and expecting the same set of key-values on the responses for APIs that use the `ActionDetails` type. Also, less hassle of augmenting types in each API response.

Reply:
I propose that we include the `hosts` property in all APIs that use `ActionDetails`. We won't use the optional param either in that case. @paul-tavares

Reply:
@ashokaditya - I'm ok with that. Lets just add it to the service and it will be return for all APIs that use it

The metadata service I think used to be expensive before we had the unified index. I think its performace is much better now, so we should be good (especially after I seeing that you are only making one call to fetch them with all IDs 👍 )